### PR TITLE
feat(static-site): parallel per-market export, India market, name under ticker

### PR DESCRIPTION
**Pull request by xang1234**
## Summary

Cumulative feature branch for the static-site multi-market expansion. Three themes:

- **India market, end to end**: universe sourcing (NSE primary CSV + BSE secondary JSON with ISIN reconciliation), calendar (XNSE + pandas_market_calendars fallback), Nifty 50 / NIFTYBEES benchmarks, `data/india-deep.csv` taxonomy, static-site export path, and inclusion in the GitHub Actions matrix. `DAILY_PRICE_SUPPORTED_MARKETS` and `WEEKLY_REFERENCE_MARKETS/SNAPSHOT_KEYS` now include `IN`.
- **GitHub-first bootstrap and daily price bundles**: new `GitHubReleaseSyncService`, `build_daily_price_bundle.py` / `sync_daily_price_bundle_from_github.py` scripts, and weekly-reference bundle publishing so fresh deployments seed their DB from GitHub Releases instead of re-fetching from providers. Docker CSV-path resolution hardened for container runs.
- **Static-site usability**: new shared `TickerCell` component that renders `symbol` + `company_name` stacked (matching the live Scan results layout), used in the daily top-scan table, the Top 10 Groups `top_symbol` cell, the Group detail modal, and the full Groups rankings table — this fixes the Asian-market readability problem where numeric codes like `0700.HK`, `2330.TW`, `005930.KS` gave no company context. Schema additions: `ConstituentStock.company_name`, `GroupRankResponse.top_symbol_name`, `GroupDetailResponse.top_symbol_name`, populated in both US and per-market ranking services. Adds a **1M** column to the daily page's Top 10 Groups panel (data was already in the payload; only the template was missing).

One backwards-compatible **behavior nuance** worth flagging: in `get_group_history`, `top_symbol_name` now resolves via the latest scan's constituents (reused from `_get_constituent_stocks`) instead of a separate `StockUniverse` query — identical under normal operation, returns `None` only in the edge case where the ranking's top_symbol has since been delisted/dropped from the current constituents.

## Test plan

- [ ] Backend unit tests pass: `cd backend && pytest tests/unit/` (note: one pre-existing `test_compute_breadth_metrics_uses_full_history_for_shifted_ranges` failure is tracked under beads `yl2h` and is unrelated to this branch).
- [ ] Frontend lint + tests: `cd frontend && npm run lint && npm run test:run && npm run build`.
- [ ] CI workflow dry-run via `workflow_dispatch` on the branch — confirm all 5 market jobs (US, HK, IN, JP, TW) complete, the combine step merges artifacts, and the deployed Pages preview shows India in the market selector.
- [ ] Visual check (`npm run dev`):
  - [ ] Market switcher includes **India**.
  - [ ] **Top 10 Groups** panel on the daily page shows `1W | 1M | Top Stock` columns.
  - [ ] **Top Scan Candidates**, **Group detail** constituents, and full **Groups** rankings show the company name under the ticker.
  - [ ] For HK / JP / TW, numeric-ticker rows now render a recognizable company name underneath.
- [ ] Prereqs before first India CI cron run:
  - [ ] Publish a `fundamentals_v1_in` weekly-reference bundle so the "Download weekly reference bundle" step can seed the DB.
  - [ ] Run a one-off 6-month US group-ranking backfill (`backfill_group_rankings_optimized(today-180d, today)`) so `rank_change_3m` / `rank_change_6m` columns populate immediately for US (tracked in beads `ctkh`). India's `rank_change_6m` will naturally fill in as feature runs accumulate.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/xang1234/stock-screener/pull/100" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Release Notes

* **New Features**
  * Added India market support to the build and ranking pipeline.
  * Enhanced ticker displays to show company names alongside stock symbols in Top Scan Candidates, Top 10 Groups, and constituent stocks tables.
  * Added 1-month rank change column to the Top 10 Groups table.

### CODE REVIEW

File: .github/workflows/static-site.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        market: [US, HK, JP, TW]
+        market: [US, HK, IN, JP, TW]
```
[the remark this answers is not in the document]
Comment (xang1234) [reply .github/workflows/static-site.yml]: Addressed in `.github/workflows/static-site.yml` by removing `IN` from the static-site matrix. I verified the current `weekly-reference-data` release still lacks `weekly-reference-latest-in.json`, and the legacy fallback manifest points to a US-only bundle, so leaving `IN` enabled here would be unsafe. Follow-up is tracked in `StockScreenClaude-s0xf` before the India job is re-enabled.

File: backend/app/services/daily_price_bundle_service.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class DailyPriceBundleService:
     DAILY_PRICE_MANIFEST_SCHEMA_VERSION = "daily-price-manifest-v1"
     DAILY_PRICE_RELEASE_TAG = "daily-price-data"
     DAILY_PRICE_BAR_PERIOD = "2y"
-    DAILY_PRICE_SUPPORTED_MARKETS: tuple[str, ...] = ("US", "HK", "JP", "TW")
+    DAILY_PRICE_SUPPORTED_MARKETS: tuple[str, ...] = ("US", "HK", "IN", "JP", "TW")
```
[the remark this answers is not in the document]
Comment (xang1234) [reply backend/app/services/daily_price_bundle_service.py:33]: Not changing this in this PR because India group rankings and the `/v1/groups` India surface are still intentionally staged behind published-run readiness. The repo does not yet publish India weekly-reference assets, and the rollout is already tracked in `StockScreenClaude-l0i3`, which now depends on `StockScreenClaude-s0xf`. Impact is limited to `/v1/groups?market=IN`, which should continue returning 400 until the India group pipeline is actually published.
